Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "RobotImporter/URDF/URDFPrefabMaker.h"
+#include "Frame/ROS2FrameComponent.h"
+#include "RobotControl/ROS2RobotControlComponent.h"
+#include "RobotImporter/URDF/CollidersMaker.h"
+#include "RobotImporter/URDF/PrefabMakerUtils.h"
+
+#include "RobotImporter/RobotImporterWidget.h"
+
+#include <AzCore/IO/FileIO.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzToolsFramework/API/EditorAssetSystemAPI.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <PhysX/MeshAsset.h>
+
+#include <regex> // TODO - we are currently replacing package:// with an absolute path
+
+namespace ROS2
+{
+    URDFPrefabMaker::URDFPrefabMaker(const AZStd::string& modelFilePath, urdf::ModelInterfaceSharedPtr model, AZStd::string prefabPath)
+        : m_model(model)
+        , m_visualsMaker(modelFilePath, model->materials_)
+        , m_collidersMaker(new CollidersMaker(modelFilePath))
```

Comment:
maybe make m_collidersMaker a AZStd::unique_ptr instead of a shared_ptr? And use AZStd::make_unique<> instead of new

Reply:
Fixing that needed to implement move constructor to CollidersMaker. Please take a look at  [that](https://github.com/RobotecAI/o3de-ros2-gem/commit/aefb1e49b84b5420155e5fac47315bd446505430#diff-1d0a02d3c5cc965628b4c5487fae26cf3de67ad9d6025b0631336ca7869a2383R37)

Reply:
cc @adamdbrw

Reply:
Or another option is to make m_collidersMaker just a member
```
CollidersMaker m_collidersMaker;
```
and then it could be initialized like
```
        , m_visualsMaker(modelFilePath, model->materials_)
        , m_collidersMaker(modelFilePath)
```